homeassistant/components/freedns.py

```diff
@@ -0,0 +1,115 @@
+"""
+Integrate with FreeDNS Dynamic DNS service at freedns.afraid.org.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/freedns/
+"""
+import asyncio
+from datetime import timedelta
+import logging
+
+import aiohttp
+from aiohttp.hdrs import USER_AGENT
+import async_timeout
+import voluptuous as vol
+
+from homeassistant.const import (CONF_URL, CONF_TIMEOUT, CONF_ACCESS_TOKEN)
+from homeassistant.helpers.aiohttp_client import SERVER_SOFTWARE
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'freedns'
+
+# We should set a dedicated address for the user agent.
+EMAIL = 'hello@home-assistant.io'
+
+DEFAULT_INTERVAL = timedelta(minutes=10)
+DEFAULT_TIMEOUT = 10
+
+UPDATE_URL = 'https://freedns.afraid.org/dynamic/update.php'
+HA_USER_AGENT = "{} {}".format(SERVER_SOFTWARE, EMAIL)
+
+CONF_UPDATE_INTERVAL = 'update_interval'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Exclusive(CONF_URL, DOMAIN): cv.string,
+        vol.Exclusive(CONF_ACCESS_TOKEN, DOMAIN): cv.string,
+        vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+        vol.Optional(CONF_UPDATE_INTERVAL, default=DEFAULT_INTERVAL): vol.All(
+            cv.time_period, cv.positive_timedelta),
+
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Initialize the FreeDNS component."""
+    url = config[DOMAIN].get(CONF_URL)
+    auth_token = config[DOMAIN].get(CONF_ACCESS_TOKEN)
+    timeout = config[DOMAIN].get(CONF_TIMEOUT)
+    update_interval = config[DOMAIN].get(CONF_UPDATE_INTERVAL)
+
+    session = hass.helpers.aiohttp_client.async_get_clientsession(False)
```

Comment:
Please don't disable the SSL verification. That is only allow for internal network stuff

Reply:
that will be called: securityhole :) On a public service, SSL verification should be not disabled or you don't need use SSL.